src/Core/Indexes/Octree.cs

```diff
@@ -70,17 +70,40 @@ protected new void CreateChildren()
         /// </summary>
         /// <param name="envelope">The maximum indexed region.</param>
         public Octree(Envelope envelope)
-            : base(envelope)
         {
+            this.root = new OctreeNode(envelope);
         }
 
         /// <summary>
         /// Initializes a new instance of the <see cref="Octree" /> class.
         /// </summary>
         /// <param name="geometries">The geometries to add to the tree.</param>
         public Octree(IEnumerable<IBasicGeometry> geometries)
-            : base(geometries)
         {
+            Envelope bound = Envelope.FromEnvelopes(geometries.Select(geometry => geometry.Envelope));
+            this.root = new OctreeNode(bound);
+
+            this.Add(geometries);
+        }
+
+        /// <summary>
+        /// Clears all geometries from the index.
+        /// </summary>
+        public new void Clear()
+        {
+            this.root = new OctreeNode(this.root.Envelope);
+        }
+
+        /// <summary>
+        /// Creates a new tree based on an unindexed geometry.
+        /// </summary>
+        /// <param name="geometry">The geometry.</param>
+        protected new void CreateNew(IBasicGeometry geometry)
```

Comment:
With the `new` keyword, you are hiding the `CreateNew` method from `QuadTree`, instead of overriding it. This can easily lead to programming errors later.

The `CreateNew` method should be defined virtual instead in `QuadTree` and overriden in `Octree`.